frame/bags-list/src/benchmarks.rs

```diff
@@ -0,0 +1,97 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Benchmarks for the bags list pallet.
+
+use super::*;
+use crate::list::List;
+use frame_benchmarking::{account, whitelisted_caller};
+use frame_election_provider_support::VoteWeightProvider;
+use frame_support::{assert_ok, traits::Get};
+use frame_system::RawOrigin as SystemOrigin;
+
+frame_benchmarking::benchmarks! {
+	rebag {
+		// An expensive case for rebag-ing:
+		//
+		// - The node to be rebagged should exist as a non-terminal node in a bag with at least
+		//   2 other nodes so both its prev and next are nodes that will need be updated
+		//   when it is removed.
+		// - The destination bag is not empty, because then we need to update the `next` pointer
+		//   of the previous node in addition to the work we do otherwise.
+		//
+		// NOTE: another expensive case for rebag-ing is for a terminal node, because in that case
+		// the tail node of each back will need to be updated and both bags will need to be read and
+		// written to storage.
```

Comment:
why not make a benchmark for both?

Reply:
Added another benchmark case 2f18113f7fe6e577ab91cfe39b8fd45a86ec7890